[2024-11-23T18:19:26.712+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-11-23T18:19:26.829+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: fraud_detection.process_transaction manual__2024-11-23T18:18:42.285049+00:00 [queued]>
[2024-11-23T18:19:26.873+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: fraud_detection.process_transaction manual__2024-11-23T18:18:42.285049+00:00 [queued]>
[2024-11-23T18:19:26.875+0000] {taskinstance.py:2865} INFO - Starting attempt 1 of 2
[2024-11-23T18:19:26.946+0000] {taskinstance.py:2888} INFO - Executing <Task(BranchPythonOperator): process_transaction> on 2024-11-23 18:18:42.285049+00:00
[2024-11-23T18:19:26.966+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=5080) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-11-23T18:19:26.968+0000] {standard_task_runner.py:72} INFO - Started process 5090 to run task
[2024-11-23T18:19:26.973+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'fraud_detection', 'process_transaction', 'manual__2024-11-23T18:18:42.285049+00:00', '--job-id', '503', '--raw', '--subdir', 'DAGS_FOLDER/fraud_detection_dag.py', '--cfg-path', '/tmp/tmp8xouvb6v']
[2024-11-23T18:19:26.978+0000] {standard_task_runner.py:105} INFO - Job 503: Subtask process_transaction
[2024-11-23T18:19:27.150+0000] {task_command.py:467} INFO - Running <TaskInstance: fraud_detection.process_transaction manual__2024-11-23T18:18:42.285049+00:00 [running]> on host 345e92ff26bf
[2024-11-23T18:19:27.429+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='fraud_team' AIRFLOW_CTX_DAG_ID='fraud_detection' AIRFLOW_CTX_TASK_ID='process_transaction' AIRFLOW_CTX_EXECUTION_DATE='2024-11-23T18:18:42.285049+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-11-23T18:18:42.285049+00:00'
[2024-11-23T18:19:27.436+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-11-23T18:19:27.488+0000] {baseoperator.py:405} WARNING - BranchPythonOperator.execute cannot be called outside TaskInstance!
[2024-11-23T18:19:27.532+0000] {fraud_detection_dag.py:78} INFO - Données récupérées depuis XCom: {"columns":["cc_num","merchant","category","amt","first","last","gender","street","city","state","zip","lat","long","city_pop","job","dob","trans_num","merch_lat","merch_long","is_fraud","current_time"],"index":[102097],"data":[[639046421587,"fraud_Casper, Hand and Zulauf","grocery_pos",94.9,"Dylan","Bonilla","M","2497 John Motorway Suite 922","Leetsdale","PA",15056,40.5662,-80.2099,1140,"Administrator, education","1986-06-20","e13757a29993940a7748272bb9d4fda6",40.03663,-79.837932,0,1732385961809]]}
[2024-11-23T18:19:27.533+0000] {fraud_detection_dag.py:86} INFO - Données JSON parsées avec succès
[2024-11-23T18:19:27.545+0000] {fraud_detection_dag.py:90} INFO - DataFrame créé avec 1 lignes
[2024-11-23T18:19:27.550+0000] {fraud_detection_dag.py:94} INFO - Colonnes renommées avec succès
[2024-11-23T18:19:27.551+0000] {fraud_detection_dag.py:95} INFO - Colonnes disponibles: ['cc_num', 'merchant', 'category', 'amt', 'first', 'last', 'gender', 'street', 'city', 'state', 'zip', 'lat', 'long', 'city_pop', 'job', 'dob', 'trans_num', 'merch_lat', 'merch_long', 'is_fraud', 'trans_date_trans_time']
[2024-11-23T18:19:27.559+0000] {fraud_detection_dag.py:99} INFO - Conversion de trans_date_trans_time en unix_time réussie
[2024-11-23T18:19:27.634+0000] {fraud_detection_dag.py:108} INFO - Transformation des données...
[2024-11-23T18:19:28.915+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/sklearn/ensemble/_base.py:172: UserWarning: Loky-backed parallel loops cannot be called in a multiprocessing, setting n_jobs=1
  n_jobs = min(effective_n_jobs(n_jobs), n_estimators)

[2024-11-23T18:19:28.939+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/sklearn/ensemble/_base.py:172: UserWarning: Loky-backed parallel loops cannot be called in a multiprocessing, setting n_jobs=1
  n_jobs = min(effective_n_jobs(n_jobs), n_estimators)

[2024-11-23T18:19:28.947+0000] {fraud_detection_dag.py:120} INFO - Prediction de fraude: NORMAL, Probabilité: 39.87%
[2024-11-23T18:19:29.010+0000] {fraud_detection_dag.py:126} INFO - Prédiction: NORMAL (Probabilité: 39.87%)
[2024-11-23T18:19:29.012+0000] {python.py:240} INFO - Done. Returned value was: notify_normal
[2024-11-23T18:19:29.014+0000] {branch.py:38} INFO - Branch into notify_normal
[2024-11-23T18:19:29.015+0000] {skipmixin.py:230} INFO - Following branch ('notify_normal',)
[2024-11-23T18:19:29.039+0000] {skipmixin.py:278} INFO - Skipping tasks [('notify_fraud', -1), ('skip_processing', -1)]
[2024-11-23T18:19:29.118+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-11-23T18:19:29.120+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=fraud_detection, task_id=process_transaction, run_id=manual__2024-11-23T18:18:42.285049+00:00, execution_date=20241123T181842, start_date=20241123T181926, end_date=20241123T181929
[2024-11-23T18:19:29.180+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2024-11-23T18:19:29.291+0000] {taskinstance.py:3900} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2024-11-23T18:19:29.309+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
